File: app/src/main/java/com/stathis/elmepaunivapp/database/AnnouncementsDatabase.java
package com.stathis.elmepaunivapp.database;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.stathis.elmepaunivapp.ui.announcements.model.Announcement;

@Database(entities = {Announcement.class}, exportSchema = false, version = 1)
public abstract class AnnouncementsDatabase extends RoomDatabase {

    private static AnnouncementsDatabase instance;

    @NonNull
    public static AnnouncementsDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), AnnouncementsDatabase.class, "AppAnnouncements.db")
                    .allowMainThreadQueries()
                    .fallbackToDestructiveMigration()
                    .build();
        }

        return instance;
    }

    public abstract AnnouncementsDao getAnnouncementDao();
}

File: app/src/main/java/com/stathis/elmepaunivapp/database/AnnouncementsDatabase.kt
package com.stathis.elmepaunivapp.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.stathis.elmepaunivapp.model.Announcement

@Database(entities = [Announcement::class] , version = 2, exportSchema = false)
abstract class AnnouncementsDatabase : RoomDatabase() {

    abstract fun announcementDao(): AnnouncementsDao

    companion object {
        // Singleton prevents multiple instances of database opening at the
        // same time.
        @Volatile
        private var INSTANCE: AnnouncementsDatabase? = null

        fun getDatabase(context: Context): AnnouncementsDatabase {
            // if the INSTANCE is not null, then return it,
            // if it is, then create the database
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    AnnouncementsDatabase::class.java,
                    "announcements_database"
                ).fallbackToDestructiveMigration().build()
                INSTANCE = instance
                // return instance
                instance
            }
        }
    }
}
